cpp/tensorrt_llm/kernels/cutlass_kernels/CMakeLists.txt

```diff
@@ -46,64 +49,94 @@ set(INSTANTIATION_GENERATION_DIR
 
 execute_process(
   WORKING_DIRECTORY ${GENERATE_KERNELS_SCRIPT_DIR}
-  COMMAND
-    ${Python3_EXECUTABLE} generate_kernels.py -a
-    "${CMAKE_CUDA_ARCHITECTURES_ORIG};${CMAKE_CUDA_ARCHITECTURES_NATIVE}" -o
-    ${INSTANTIATION_GENERATION_DIR}
-  OUTPUT_VARIABLE _KERNEL_GEN_OUTPUT
+  COMMAND ${Python3_EXECUTABLE} generate_kernels.py -a
+          "${CMAKE_CUDA_ARCHITECTURES_ORIG}" -o ${INSTANTIATION_GENERATION_DIR}
   RESULT_VARIABLE _KERNEL_GEN_SUCCESS)
 
-if(NOT _KERNEL_GEN_SUCCESS MATCHES 0)
+if(NOT _KERNEL_GEN_SUCCESS EQUAL 0)
   message(
     FATAL_ERROR
       "Failed to generate internal CUTLASS kernel instantiations due to ${_KERNEL_GEN_SUCCESS}."
   )
 endif()
 
-file(GLOB_RECURSE INSTANTIATIONS_GENERATED ${INSTANTIATION_GENERATION_DIR}/*.cu)
+function(process_target target_name enable_hopper enable_blackwell)
 
-string(STRIP "${_KERNEL_GEN_OUTPUT}" _KERNEL_GEN_OUTPUT)
+  if(${enable_hopper} AND "90" IN_LIST CMAKE_CUDA_ARCHITECTURES_ORIG)
+    # No kernels should be parsed, unless hopper is specified. This is a build
+    # time improvement
+    target_compile_definitions(${target_name} PUBLIC COMPILE_HOPPER_TMA_GEMMS)
+    target_compile_definitions(${target_name}
+                               PUBLIC COMPILE_HOPPER_TMA_GROUPED_GEMMS)
+  endif()
 
-# Sort both lists to ensure order doesn't matter
-list(SORT _KERNEL_GEN_OUTPUT)
-list(SORT INSTANTIATIONS_GENERATED)
+  if(${enable_blackwell} AND ("100" IN_LIST CMAKE_CUDA_ARCHITECTURES_ORIG
+                              OR "120" IN_LIST CMAKE_CUDA_ARCHITECTURES_ORIG))
```

Comment:
What is the point of including 120 in this if condition if it doesnt do anything

Reply:
This is some leftover that became meaningless after we open sourced these kernels. Removed.